### DISCUSSION TITLE
How to use Respect "retry-after" when the response-header Retry-After is alpha-numeric

### DISCUSSION THREAD

**Discussion by bakaito** [tilln/jmeter-retrier#2] [Q&A]
Hi, I am doing a rate-limit testing using the plugin. However, to be able to use the respect retry-after option, our response header is returned in the following format:
Retry-After: 10s

How to extract just the number and provide it to the retry-after option and retry the API call after 10 seconds? also this value is not static and is returned dynamically in the response header

**Comment by tilln** [comment]
Hi @bakaito 

As you may be aware, your format does not comply with the HTTP specification as per https://www.rfc-editor.org/rfc/rfc9110#field.retry-after

Perhaps you want to raise this with your development team first whether they really intend to deviate from the HTTP spec?

**Reply by bakaito** [reply]
Hi @tilln . Thank you for your prompt response. That's a good point. I would check with my dev team, however, considering most of our orchestrations have been already built over this API implementation, I doubt if they would consider changing it. In such a case, is it possible to handle the retry-after with this plugin?

**Reply by tilln** [reply]
> considering most of our orchestrations have been already built over this API implementation, I doubt if they would consider changing it.

I don't follow this reasoning, how this justifies implementing something that violates an open standard.
The change itself would be trivial.
I consider this a defect in the API implementation. 
Besides, API consumers would also have to deal with this. 
Lastly, the `s` provides no value or additional information than a standard header.

>  is it possible to handle the retry-after with this plugin?

Not without some code changes.

Trivial changes, though while dealing with the `s` is not a problem, the question is what other variations this plugin should cater for and how much of this should be configurable by users. 
What about spaces, e.g. `10 s`?
What about non-integer values, e.g. `3.5s`?
What about `500ms`?

**Reply by bakaito** [reply]
Hey again. we got things changed from our end. It was also not consistent across APIs where some of them have had rate limits per hour and some of them in seconds. Very much appreciated your inputs.

**Reply by tilln** [reply]
Thanks, @bakaito, for reporting back.
Happy you got it sorted.
So, now you can use this plugin as-is :⁠-⁠)
